x-pack/plugins/fleet/server/routes/setup/handlers.ts

```diff
@@ -46,8 +46,14 @@ export const fleetSetupHandler: RequestHandler = async (context, request, respon
   try {
     const soClient = context.core.savedObjects.client;
     const esClient = context.core.elasticsearch.client.asCurrentUser;
-    const body: PostIngestSetupResponse = { isInitialized: true };
-    await setupIngestManager(soClient, esClient);
+    const setupStatus = await setupIngestManager(soClient, esClient);
+    const body: PostIngestSetupResponse = {
+      isInitialized: true,
+    };
+
+    if (setupStatus.nonFatalPackageUpgradeErrors.length > 0) {
+      body.nonFatalPackageUpgradeErrors = setupStatus.nonFatalPackageUpgradeErrors;
+    }
 
```

Comment:
This is unrelated to my change, but I noticed that `preconfigurationError` was never returned here. I didn't have time to dig into it but I'm not sure this is intentional. cc @Zacqary

Reply:
I remember some discussion around what should happen on errors. I'm sure @Zacqary will follow up if needed.